.github/workflows/icon4py-qa.yml

```
@@ -13,24 +13,27 @@ jobs:
     pre-commit-icon4py-model:
         runs-on: ubuntu-latest
         steps:
-            - uses: actions/checkout@v3
-            -   name: Install system dependencies
-                run: |
-                    sudo apt-get update
-                    sudo apt-get install libboost-all-dev
-                    curl -LsSf https://astral.sh/uv/install.sh | sh
+            - uses: actions/checkout@v4
+
+            - name: Install system dependencies
+              run: |
+                  sudo apt-get update
+                  sudo apt-get install libboost-all-dev
+
             - name: Set up Python
-              uses: actions/setup-python@v4
+              uses: actions/setup-python@v5
               with:
-                  python-version: "3.10"
-                  cache: 'pip'
-                  cache-dependency-path: |
-                    **/pyproject.toml
-                    uv.lock
-            - name: Install venv and icon4py namespace packages
-              run: |
-                  uv sync --extra all
-            - name: Run checks
-              run: |
-                  source .venv/bin/activate
-                  pre-commit run
+                python-version-file: ".default-python-version"
```

As mentioned above, let's use the existing `.python-version` for now:
```suggestion
                python-version-file: ".python-version"
```

---

Ah, thanks, I completely missed that file! It wasn't showing up in my file searches because it's in `.gitignore`. Is the idea that it's common to change the version locally but that change should not be propagated into the repo, so it's better to have it in `.gitignore` to avoid accidental checkins?

---

Hmm, you're right. I had missed completely that `.python-version` is in the `.gitignore`, which doesn't make sense. I propose to rename it then to `.default-python-version`, check that everything still works and then merge.